File: ebi-mission/src/model/dto/CartProductDto.js

class CartProductDto {
  constructor(cartSn, trNo, lrtrNo, spdNm, brdNm, slPrc, estmtDlvTxt, odQty, spdNo, sitmNo, mbNo, imgJsn, regDttm) {
    this.cartSn = cartSn;
    this.trNo = trNo;
    this.lrtrNo = lrtrNo;
    this.spdNm = spdNm;
    this.brdNm = brdNm;
    this.slPrc = slPrc; 
    this.estmtDlvTxt = estmtDlvTxt;
    this.odQty = odQty;
    this.spdNo = spdNo;
    this.sitmNo = sitmNo;
    this.mbNo = mbNo;
    this.imgJsn = imgJsn;
    this.regDttm = regDttm;
  }
  
  toCart(jsonObj) {
    return new CartProductDto(jsonObj.trNo, jsonObj.spdNm, jsonObj.brdNm, jsonObj.lPrc, jsonObj.estmtDlvTxt, jsonObj.odQty, jsonObj.spdNo, jsonObj.sitmNo, jsonObj.mbNo, jsonObj.imgJsn)
  }
}

export default CartProductDto;

File: eBImission/src/main/java/com/example/eBImission/entity/dto/CartProductDto.java
package com.example.eBImission.entity.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
// Json 입력값에 대하여 Mapping 시 클래스에 선언되지 않은 property 무시
@JsonIgnoreProperties(ignoreUnknown = true)
public class CartProductDto {
    private int cartSn;
    private String trNo;
    private String spdNm;
    private String brdNm;
    private String slPrc;
    private String estmtDlvTxt;
    private int odQty;
    private String spdNo;
    private String sitmNo;
    private String mbNo;
    private ImgJsn[] imgJsn;
}
